docs/application/native/guides/machine-learning/machine-learning-inference.md

```diff
@@ -298,6 +298,135 @@ Followings are the available elements:
     char pipeline[] = "videotestsrc is-live=true ! videoconvert ! tensor_converter ! output-selector name=outs outs.src_0 ! tensor_sink name=sink0 async=false outs.src_1 ! tensor_sink name=sink1 async=false"
     ```
 
+- **If**
+
+    The if element (tensor_if) controls its src-pad based on the values (other/tensor(s)) of its sink-pad.
+
+    For example, you may skip frames if there is no object detected with high confidence.
+
+    The input/output stream data type is either ```other/tensor``` or ```other/tensors```.
+
+    - **Properties**
+      - compared-value: Compared Value, operand 1 (from input tensor(s)).
+        * A_VALUE:  Decide based on a single scalar value
+        * TENSOR_AVERAGE_VALUE: Decide based on a average value of a specific tensor
+        * CUSTOM: Decide based on a user defined callback
+
+      - compared-value-option: Specify an element of a value of nth tensor or pick tensor(s)
+        * [C][W][H][B],n (e.g., 0:1:2:3,0 means [0][1][2][3] value of first tensor)
+        * nth tensor, nth tensor, ...
+
+      - supplied-values: Supplied value, operand 2 (from the properties).
+        * SV
+        * SV1,SV2 (for RANGE operators)
+
+      - operator: Comparison operator comparing compared-value (CV) and supplied-value (SV)
+        * EQ: Check if CV == SV
+        * NEQ: Check if CV != SV
+        * GT: Check if CV > SV
+        * GE: Check if CV >= SV
+        * LT: Check if CV < SV
+        * LE: Check if CV <= SV
+        * RANGE_INCLUSIVE: Check if SV1 <= CV and CV <= SV2
+        * RANGE_EXCLUSIVE: Check if SV1 < CV and CV < SV2
+        * NOT_IN_RANGE_INCLUSIVE: Check if CV < SV1 or SV2 < CV
+        * NOT_IN_RANGE_EXCLUSIVE: Check if CV <= SV1 or SV2 <= CV
+
+      - then: Action if the result is TRUE
+        * PASSTHROUGH: The element will not make changes to the buffers, buffers are pushed straight through.
+        * SKIP: Do not generate output frame (frame skip)
+        * TENSORPICK: Choose nth tensor (or tensors) among input tensors
+            ```
+            [ tensor 0 ]
+            [ tensor 1 ]  ->   tensor if    ->    [ tensor 0 ]
+            [ tensor 2 ]    (tensorpick 0,2)      [ tensor 2 ]
+            input tensors                         output tensors
+            ```
+
+      - then-option: Option for TRUE action
+        * nth tensor, nth tensor, ... (for TENSORPICK option)
+
+      - else: Action if the result is FALSE
+        * PASSTHROUGH: The element will not make changes to the buffers, buffers are pushed straight through.
+        * SKIP: Do not generate output frame (frame skip)
+        * TENSORPICK: Choose nth tensor (or tensors) among input tensors
+
+      - else_option: Option for FALSE action
+        * nth tensor,nth tensor,... (for TENSORPICK option)
+
```

Comment:
@gichan-jang
Request you to provide the API reference and where from can I study these properties , that will help me review this in better way.